client/server/pages/index.js

```diff
@@ -719,15 +719,19 @@ function wpcomPages( app ) {
 		if ( ! req.context.isLoggedIn ) {
 			const queryFor = req.query?.for;
 			const ref = req.query?.ref;
+			const coupon = req.query?.coupon;
 
 			if ( queryFor && 'jetpack' === queryFor ) {
 				res.redirect(
 					'https://wordpress.com/wp-login.php?redirect_to=https%3A%2F%2Fwordpress.com%2Fplans'
 				);
 			} else {
 				const pricingPage = 'https://wordpress.com/pricing/';
-				const refQuery = ref ? `?ref=${ ref }` : '';
-				const pricingPageUrl = localizeUrl( `${ pricingPage }${ refQuery }`, locale );
+				const queryString = stringify( { ref, coupon } );
```

Comment:
`Non-blocking`:

We currently support two price reduction query params in `/plans` -- `coupon` and `discount`. When I last checked, the latter was used by email marketing campaigns. I'm unsure when this is changing, but for the time being, could we support both here as well?

cc @bubel would it be possible to discuss switching to use of the `coupon` query param in email marketing urls moving forward? Our onboarding flows exclusively use it, and now `/plans` supports it as well.

Reply:
A bit more context here https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/controller.jsx#L35-L41